test/api/application/services/test_authorisation_service.py

```diff
@@ -117,341 +157,661 @@ def test_raises_error_when_browser_makes_request_and_no_user_token_provided(self
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["READ_SENSITIVE"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["WRITE_PUBLIC"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["WRITE_PRIVATE"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["WRITE_SENSITIVE"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["DELETE_PUBLIC"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["DELETE_PRIVATE"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["DELETE_SENSITIVE"], ["DELETE"]),
-
-        # Token with ALL permission
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["READ_ALL"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["READ_ALL"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["READ_ALL"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["WRITE_ALL"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["WRITE_ALL"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["WRITE_ALL"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["DELETE_ALL"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["DELETE_ALL"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.SENSITIVE, ["DELETE_ALL"], ["DELETE"]),
-
-        # Higher sensitivity levels imply lower ones
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["READ_SENSITIVE"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["READ_SENSITIVE"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["READ_PRIVATE"], ["READ"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["WRITE_SENSITIVE"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["WRITE_SENSITIVE"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["WRITE_PRIVATE"], ["WRITE"]),
-        ("domain", "dataset", SensitivityLevel.PRIVATE, ["DELETE_SENSITIVE"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["DELETE_SENSITIVE"], ["DELETE"]),
-        ("domain", "dataset", SensitivityLevel.PUBLIC, ["DELETE_PRIVATE"], ["DELETE"]),
-
-        # Standalone scopes (no domain or dataset, different type of action)
-        (None, None, None, ["ADD_CLIENT"], ["ADD_CLIENT"]),
-        (None, None, None, ["ADD_SCHEMA"], ["ADD_SCHEMA"]),
-    ])
-    def test_matches_permissions(self,
-                                 mock_s3_adapter,
-                                 domain: str,
-                                 dataset: str,
-                                 sensitivity: SensitivityLevel,
-                                 token_scopes: List[str],
-                                 endpoint_scopes: List[str]):
+    @pytest.mark.parametrize(
+        "domain, dataset, sensitivity, token_scopes, endpoint_scopes",
+        [
```

Comment:
No specific suggestion, observing that this feels really messy; any way to simplify?

Reply:
Just black changes